packages/lit-html/src/lit-html.ts

```diff
@@ -999,10 +996,27 @@ class Template {
 
   // Overridden via `litHtmlPolyfillSupport` to provide platform support.
   /** @nocollapse */
-  static createElement(html: TrustedHTML, _options?: RenderOptions) {
-    const el = d.createElement('template');
-    el.innerHTML = html as unknown as string;
-    return el;
+  static createElement(
+    html: TrustedHTML,
+    _options?: RenderOptions | undefined,
+    type?: ResultType
+  ) {
+    if (this.parser === undefined) {
+      this.parser = new DOMParser();
+    }
+    const doc = this.parser.parseFromString(
+      '<body>' + (html as unknown as string) + '</body>',
```

Comment:
This concerns me. Maybe all element types work in `<body>`, but they definitely do in `<template>`

Reply:
Yeah I don't like it either but couldn't think of a better work around. I noticed test failures without this and remembered I had to do the same thing for labs/testing here: https://github.com/lit/lit/blob/cabe72a863a5de14a4bbca384374db748dd9b4c5/packages/labs/testing/src/lib/fixtures/ssr-fixture.ts#L85-L94

Any top level comment gets moved outside the `<body>` tag by the DOM Parser.
![image](https://github.com/lit/lit/assets/40413829/948eefab-0fd9-4640-b554-dd79b48bad7b)

Reply:
Might be relevant: https://html.spec.whatwg.org/multipage/parsing.html#parsing-main-inbody